### DISCUSSION TITLE
Switching to Vite

### DISCUSSION THREAD

**Discussion by swalker326** [epicweb-dev/epic-stack#510] [Ideas]
I'm sure its too soon to tell, but the vite support looks lightning fast. I've always been a huge fan of vite and would like to get some input on moving over to it. I know there is a issue #363 which talks about the speed of dev server start up. The video from remix looked just wild. 

https://www.youtube.com/watch?v=B_vIp4xETl4

Thoughts?

**Comment by kentcdodds** [comment]
Yes, I'm definitely interested!

I'm working on updating my own site first to get a feel for things. I've hit a couple of small bumps. I'll get through those and get around to testing out vite on the Epic Stack. I welcome a PR as well. I think it'd be better if we wait until they remove the "not recommended for production" message in the console when you run the vite build though 😅

I do think this will speed up the server startup time. Still we've got about 1.5 seconds that are our responsibility on that so it won't quite be instant. I'm also interested in swapping from express to fastify so maybe that'll improve things on that end.

Lots to do 😅

**Reply by swalker326** [reply]
Unless someone else jumps on it, I'll get a pr this week.

**Reply by stephen776** [reply]
I’m curious about the Fastify vs Hono decision. Is there a discussion somewhere?

**Reply by isherbuilds** [reply]
I thinking it's mainly because there is no helmet alternative for hono yet

**Reply by stephen776** [reply]
There is now as far as I understand. 

https://github.com/honojs/hono/pull/1341

Maybe this isn’t sufficient? I’m interested in Hono for both speed and potential to support bun at some point.

official docs: https://hono.dev/middleware/builtin/secure-headers

I’ll concede I am out of the loop on the merits of one over the other. So perhaps Fastify is ultimately the better option. That’s why I’m curious

**Reply by swalker326** [reply]
Just a heads-up: Fastify's now playing nice with Bun, as you can see over here: https://github.com/oven-sh/bun/issues/313.

On the subject of Hono, it's definitely the newcomer and seems a bit untested compared to others. No disrespect to it – innovation is key, and I'm genuinely excited about what Bun brings to the table. But when you stack Hono against Fastify, you've got Hono (2022) quickly rolling out three major updates in about a year. Fastify, on the other hand, has taken a more measured approach since its debut in 2016, with four major updates across eight years. That's a brisk pace for Hono, which might have led to users needing to update more often than they'd like.

Looking at it from Epic Stack's standpoint, leaning on a newer, less tested platform might not send the right message. Fastify's been around longer and has the track record to show for it, which could be a safer bet for delivering solid, reliable solutions. Just my two cents though.

**Reply by kentcdodds** [reply]
Let's discuss this here: https://github.com/epicweb-dev/epic-stack/discussions/511
